concent_api/core/validation.py

```diff
@@ -106,6 +106,23 @@ def validate_list_of_identical_task_to_compute(list_of_task_to_compute: List[mes
     return True
 
 
+def validate_list_of_identical_report_computed_task(list_of_report_computed_task: List[message.tasks.ReportComputedTask]):
+    assert isinstance(list_of_report_computed_task, list)
+    assert all([isinstance(report_computed_task, message.tasks.ReportComputedTask) for report_computed_task in list_of_report_computed_task])
+
+    if len(list_of_report_computed_task) <= 1:
+        return False
+
+    base_report_computed_task = list_of_report_computed_task[0]
+
+    for report_computed_task in list_of_report_computed_task:
```

Comment:
There's no need to start from the first element because then you're comparing it with itself. This would work just as fine if you started from the second one:
``` python
for report_computed_task in list_of_report_computed_task[1:]:
```
as a bonus this works even if there's just one message on the list.

There's also a very easy way to create a version that does not need any special cases, even for zero:
```
for message1, message2 in zip(list_of_report_computed_task, list_of_report_computed_task[1:]):
    # Compare message1 and message 2
```

Reply:
We can't do it this way becuase sometimes we comparing more than 2 messages.

Reply:
Try it. It works on more than two messages just fine.

`zip()` combines two lists into a list of pairs. Say, we have messages A, B, C, D, E. This code will give you pairs (A, B), (B, C), (C, D), (D, E). If all of these pairs contain identical messages, it's mathematically guaranteed that all the messages on the list are identical.

Reply:
@cameel I am thinking about this one, isn't it also guaranteed if we take element A and compare it with all following elements ? Also wouldn't, it be quicker ?

Reply:
Yes, it's guaranteed too. I wasn't saying that the current method does not work. I only pointed out that you can do this without the extra `if`s and without singling out the first element.

Also, I'm not saying that this method is overall the best and that we have to use it. That's for your consideration. It's shorter but that's not the only criterion. I'm only responding to a comment that it would not work. For me the weak point of my approach is that it may not obvious what `zip()` does when lists are not of the same length (it ignores the extra items).

As for which one executes faster - I think that your method might be faster but not significantly. It's a difference of a few assignments (times the number of items). And given that we're doing that for very short lists of constant length, readability matters more. Please choose the one you think is easier to read and understand, not the one that's faster.